internal/ingest/httpingest/httpingest.go

```diff
@@ -16,7 +17,8 @@ import (
 type Handler struct {
 	Collector Collector
 
-	Logger *slog.Logger
+	Logger  *slog.Logger
+	Context context.Context
```

Comment:
I think it would be better to pass the context to `processEvent` as the first argument and use `r.Context` in both `processBatchRequest`, and `processSingleRequest`.